crates/rpc/rpc-eth-api/src/types.rs

```diff
@@ -26,11 +26,14 @@ pub trait EthApiTypes: Send + Sync + Clone {
         HeaderResponse = reth_rpc_types::Header,
         ReceiptResponse = AnyTransactionReceipt,
     >;
+    /// Conversion methods for transaction RPC type.
+    type TransactionCompat: Send + Sync + Clone + fmt::Debug;
```

Comment:
if we can't have constructors of the types in the trait, then we will need their constructors as separate ATs, like is done here

Reply:
why can't we solve this on the impl level, the impl should be responsible for creating the rpc object.

this trait has the same limitation as described in https://github.com/alloy-rs/alloy/pull/1173#pullrequestreview-2286039569